test/s3/s3.test.ts

```diff
@@ -183,4 +183,198 @@ describe('S3 Flow tests', () => {
       },
     });
   });
-});
\ No newline at end of file
+
+  test('Destination with catalog', () => {
+    const stack = new Stack(undefined, undefined);
+
+    const bucket = Bucket.fromBucketName(stack, 'TestBucket', 'test-bucket');
+
+    const source = new S3Source({
+      bucket: bucket,
+      prefix: 'source-prefix',
+      format: {
+        type: S3InputFileType.JSON,
+      },
+    });
+
+    const destination = new S3Destination({
+      location: { bucket: bucket },
+      catalog: {
+        database: new Database(stack, 'TestDb', {
+          databaseName: 'testdb',
+        }),
+        tablePrefix: 'tablePrefix',
+      },
+    });
+
+    new OnDemandFlow(stack, 'testFlow', {
+      source: source,
+      destination,
+      mappings: [Mapping.mapAll({ exclude: ['field1', 'field2'] })],
+    });
+
+    const template = Template.fromStack(stack);
+
+    template.resourceCountIs('AWS::S3::Bucket', 0);
+    template.resourceCountIs('AWS::AppFlow::Flow', 1);
+    template.resourceCountIs('AWS::IAM::Role', 1);
```

Comment:
It would be nice to add a test on the role properties to ensure it's scoped to the database, schema, and tables.

Reply:
Good call.

Reply:
Please take a look at the updated test. Now there is a strict assertion on the actions and resources for the IAM Policy.